src/main/java/ch/njol/skript/bukkitutil/TeleportFlags.java

```diff
@@ -0,0 +1,79 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.bukkitutil;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import io.papermc.paper.entity.TeleportFlag;
+
+import org.bukkit.Location;
+import org.bukkit.entity.Entity;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * A utility interface to access the Entity::teleport with TeleportFlag vararg in Paper 1.19+.
+ */
+@FunctionalInterface
+public interface TeleportFlags {
+
+	public enum SkriptTeleportFlags {
+		RETAIN_OPEN_INVENTORY(TeleportFlag.EntityState.RETAIN_OPEN_INVENTORY),
+		RETAIN_PASSENGERS(TeleportFlag.EntityState.RETAIN_PASSENGERS),
+		RETAIN_VEHICLE(TeleportFlag.EntityState.RETAIN_VEHICLE),
+		RETAIN_DIRECTION(),
+		RETAIN_PITCH(TeleportFlag.Relative.PITCH),
+		RETAIN_YAW(TeleportFlag.Relative.YAW),
+		RETAIN_X(TeleportFlag.Relative.X),
+		RETAIN_Y(TeleportFlag.Relative.Y),
+		RETAIN_Z(TeleportFlag.Relative.Z);
+
+		TeleportFlag flag;
+
+		SkriptTeleportFlags() {}
+
+		SkriptTeleportFlags(TeleportFlag flag) {
+			this.flag = flag;
+		}
+
+	}
+
+	void teleport(
+		@NotNull Entity entity, @NotNull Location location, @NotNull TeleportFlag... flags
+	);
+
+	static void teleport(
```

Comment:
i don't know that it really makes sense for the flags class to have the teleportation logic

Reply:
You can't reference it elsewhere due to static references and TeleportFlag not existing in versions lower than 1.19 which is why this is a functional interface and in it's own class.

Reply:
maybe a functional interface isn't the best way to implement this